### ISSUE TITLE
eval and CSP

### ISSUE THREAD

**Issue by firien** [bpampuch/pdfmake#816]
our CSP does not allow `eval` and we recently updated to v1.23 and the `createPdf` function is no longer being set on the `pdfMake` object, because of `new Function` calls.

We have been using this library since March 2016 so I believe we we're using v1.20 and this wasn't an issue before.

It is specifically complaining about the line below, but I see other `new Function` calls scattered about. I only see them in the build directory, which seems to indicate these are created when grunt (or gulp?) is "compiling" the source. Are there settings in the build process to circumvent these evals?

```js
this.relativeToGetter = new Function('ctx', "return ctx." + this.options.relativeTo);
```

**Comment by liborm85** [comment]
Pdfmake v0.1.20 using an earlier version [pdfkit library](https://github.com/devongovett/pdfkit). In pdfmake v0.21+ is a newer version library (v0.8.0), this has a dependency on [restructure library](https://github.com/devongovett/restructure) where eval is used, see [source code](https://github.com/devongovett/restructure/blob/9d505ca03a9e00add6cde0e8bf00296f70776c8e/src/Pointer.coffee#L11).
Pdfmake can not solve this problem, it must be resolved in restructure library. You can try to create an issue in [restructure library](https://github.com/devongovett/restructure).

**Comment by firien** [comment]
got it. thank you.

**Comment by liborm85** [comment]
Issue fixed in version 0.2.0 - https://github.com/bpampuch/pdfmake/issues/2282.
